common/sampling.cpp

```diff
@@ -1,7 +1,7 @@
 #include "sampling.h"
 
 struct llama_sampling_context * llama_sampling_init(const struct llama_sampling_params & params) {
-    struct llama_sampling_context * result = new llama_sampling_context();
+    auto * result = new llama_sampling_context();
```

Comment:
C-like vs C++-like. Also here I vastly prefer `auto * result = ...` to `struct llama_sampling_context * = ...`, but not everybody does. Obviously the `struct` keyword is not necessary in C++ code, so one might hypothesize that this has been written by someone who is more familiar with C than with C++. But `git blame` shows @ggerganov being the last person to touch this line. Knowing he is equally familiar with C and C++, the only possible conclusion is that he preferred it that way. Why would you go and change it?

Reply:
I don't have a strong preference and/or desire to argue here, the main argument here is to minimize the amount of code.